File: webapp/magazzino/viewsets/prodotto.py
from django_filters.rest_framework import DjangoFilterBackend
from rest_framework import filters, permissions, viewsets

from ..models.prodotto import Prodotto
from ..serializers import ProdottoPolymorphicSerializer


class ProdottoViewSet(viewsets.ModelViewSet):
    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
    queryset = Prodotto.objects.all()
    serializer_class = ProdottoPolymorphicSerializer
    filter_backends = [DjangoFilterBackend, filters.SearchFilter]
    search_fields = ["nome", "descrizione_breve", "descrizione"]
    filterset_fields = [
        "categoria",
        "categoria_secondaria",
        "url",
        "sito",
        "servizio",
        "categoria__url",
    ]

File: clients/www/src/models/prodotto.ts
export default interface Prodotto {
  id?: number;
  sito?: boolean;
  servizio?: boolean;
  nome?: string;
  descrizione_breve?: string;
  descrizione?: string;
  url?: string;
  prezzo?: string;
  prezzo_a_partire?: boolean;
  prezzo_offerta?: string;
  immagini?: { thumbnail: boolean; file: string }[];
  resourcetype?: string;
  varianti?: Prodotto[];
}
